### ISSUE TITLE
Long app version names cause MS Hockeyapp to not install the app

### ISSUE THREAD

**Issue by master-lincoln** [microsoft/appcenter#759]
**What App Center service does this affect?**
Android App Center  (Hockeyapp) version 2.0.5 (18)

**Describe the bug**
Long version names cause the client to not install the app.

**To Reproduce**
1. Upload an app with a long version name (here 100 characters)
2. Open MS Hockeyapp in Android and search for your app entry
3. Click `Download` and observer the spinner progress and then stay a full circle

**Expected behavior**
After clicking download in Hockeyapp the app should download and then prompt me to install.

**Smartphone:**
 - Device: Samsung Galaxy S8 (SM-G950F)
 - OS: Android 9

**Additional context**
Android `versionName` is set to 
```
0.54.24-bugfixAPP-1234-new-account-isn-t-being-created-after-updating-app-to-new-version.0-SNAPSHOT
```

ADB logcat example output:
```
07-23 11:17:11.031 26224 26224 E Event   : android.content.ActivityNotFoundException: No Activity found to handle Intent { act=android.intent.action.INSTALL_PACKAGE dat=content://com.microsoft.hockeyapp.testerapp.provider/external_files/Download/Fizzled-Feature-54024-(0.54.24-bugfixAPP-1234-new-account-isn-t-being-created-after-updating-app-t flg=0x1 }
07-23 11:17:11.031 26224 26224 E Event   : 	at android.app.Instrumentation.checkStartActivityResult(Instrumentation.java:2018)
07-23 11:17:11.031 26224 26224 E Event   : 	at android.app.Instrumentation.execStartActivity(Instrumentation.java:1673)
07-23 11:17:11.031 26224 26224 E Event   : 	at android.app.Activity.startActivityForResult(Activity.java:4689)
07-23 11:17:11.031 26224 26224 E Event   : 	at android.support.v4.app.i.startActivityForResult(Unknown Source:10)
07-23 11:17:11.031 26224 26224 E Event   : 	at android.app.Activity.startActivityForResult(Activity.java:4647)
07-23 11:17:11.031 26224 26224 E Event   : 	at android.support.v4.app.i.startActivityForResult(Unknown Source:10)
07-23 11:17:11.031 26224 26224 E Event   : 	at android.app.Activity.startActivity(Activity.java:5008)
07-23 11:17:11.031 26224 26224 E Event   : 	at android.app.Activity.startActivity(Activity.java:4976)
07-23 11:17:11.031 26224 26224 E Event   : 	at net.hockeyapp.client.app.appdetails.AppVersionDetailFragment.c(Unknown Source:50)
07-23 11:17:11.031 26224 26224 E Event   : 	at net.hockeyapp.client.app.appdetails.AppVersionDetailFragment.onEventMainThread(Unknown Source:66)
07-23 11:17:11.031 26224 26224 E Event   : 	at java.lang.reflect.Method.invoke(Native Method)
07-23 11:17:11.031 26224 26224 E Event   : 	at a.a.a.c.a(Unknown Source:12)
07-23 11:17:11.031 26224 26224 E Event   : 	at a.a.a.c.a(Unknown Source:65)
07-23 11:17:11.031 26224 26224 E Event   : 	at a.a.a.c.a(Unknown Source:42)
07-23 11:17:11.031 26224 26224 E Event   : 	at a.a.a.c.a(Unknown Source:26)
07-23 11:17:11.031 26224 26224 E Event   : 	at a.a.a.c.d(Unknown Source:50)
07-23 11:17:11.031 26224 26224 E Event   : 	at a.a.a.c.e(Unknown Source:13)
07-23 11:17:11.031 26224 26224 E Event   : 	at net.hockeyapp.client.download.DownloadManagerAppDownloadService$1.onReceive(Unknown Source:105)
```
(App name changed for privacy reasons)

**Comment by ranterle** [comment]
Hi there, thanks for reporting this - this looks like a bug and we are actively investigating this.

**Comment by ranterle** [comment]
@master-lincoln We just released version 2.0.7 of the Android App Center (HockeyApp) app to the Play Store which addresses this issue. Can you confirm that downloading and installing builds with long version names now works for you?
On caveat is that we now shorten individual components of the downloaded file name, so they might be a little harder to tell apart later from within your download manager if you look at it.

**Comment by master-lincoln** [comment]
I can confirm that the problem is gone in 2.0.7.
We can live with the truncation.
